sdks/wasm/ProxyDriver/Program.cs

```diff
@@ -1,28 +1,180 @@
 ﻿using System;
+using System.Net.Http;
 using System.Collections.Generic;
 using System.IO;
 using System.Linq;
+using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore;
 using Microsoft.AspNetCore.Hosting;
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.Logging;
+using System.Diagnostics;
+using Newtonsoft.Json.Linq;
 
 namespace WsProxy
 {
     public class Program
     {
-        public static void Main(string[] args)
-        {
-		    var host = new WebHostBuilder()
-		        .UseSetting(nameof(WebHostBuilderIISExtensions.UseIISIntegration), false.ToString())
+		public static void Main(string[] args)
+		{
+			var host = new WebHostBuilder()
+				.UseSetting(nameof(WebHostBuilderIISExtensions.UseIISIntegration), false.ToString())
+				.ConfigureAppConfiguration (config => config.AddCommandLine (args))
 		        .UseKestrel()
 		        .UseContentRoot(Directory.GetCurrentDirectory())
 		        .UseStartup<Startup>()
 		        .UseUrls("http://localhost:9300")
 		        .Build();
-
-		    host.Run();
+			if (args.Length > 0 && args [0].StartsWith("/Debug=",StringComparison.OrdinalIgnoreCase)) {
+				var path = args [0].Substring ("/Debug=".Length);
+				var elems = path.Split ('/');
+				var x = new ChromeDebuggingSession (elems [elems.Length - 1]);
+				x.Start ();
+			}
+			host.Run();
         }
     }
+
+	public class TestHarnessProxy {
+		static IWebHost host;
+
+		public static void Start (string chrome_path, string app_path, string page_path) {
+			lock (typeof (TestHarnessProxy)) {
+				if (host != null)
+					return;
+
+				//FIXME wtf ConfigureAppConfiguration
+				string[] args = new [] {
+					$"/ChromePath={chrome_path}",
+					$"/AppPath={app_path}",
+					$"/PagePath={page_path}",
+				};
+
+				var h = new WebHostBuilder()
+					.UseSetting(nameof(WebHostBuilderIISExtensions.UseIISIntegration), false.ToString())
+					.ConfigureAppConfiguration (config => config.AddCommandLine (args))
+			        .UseKestrel()
+			        .UseStartup<TestHarnessStartup>()
+			        .UseUrls("http://localhost:9300")
+			        .Build();
+				host = h;
+				Task.Run (() => { host.Run (); });
+
+				//FIXME implement this using socket polling so it's faster
+				Thread.Sleep (1000);
+				Console.WriteLine ("WebServer Ready!");
+			}
+		}
+	}
+
+	class ChromeDebuggingSession {
+		string page_name;
+		TaskCompletionSource<bool> all_done = new TaskCompletionSource<bool> ();
+
+		public ChromeDebuggingSession (string page_name)
+		{
+			this.page_name = page_name;
+		}
+
+		public void Start ()
+		{
+			Task.Run (() => { Launch(); });
+		}
+
+		static string GuessChromePath() {
+			string chrome_path = null;
+			if (File.Exists ("/Applications/Google Chrome Canary.app/Contents/MacOS/Google Chrome Canary"))
```

Comment:
don't duplicate these literals, seems like a typo hazard if we ever come back and add a new path or change one